app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/model/MovieProvider.kt

```diff
@@ -1,19 +1,15 @@
 package com.example.marcinwisniewski.intivetrainingapp.movielist.model
 
-import com.example.marcinwisniewski.intivetrainingapp.network.ApiFactory
+import com.example.marcinwisniewski.intivetrainingapp.network.MovieService
 import io.reactivex.Single
+import javax.inject.Inject
+import javax.inject.Singleton
 
 const val ALL_MOVIES_URL = "/filippella/Sample-API-Files/master/json/movies-api.json"
 
-class MovieProvider {
-    // TO DO : set and dependency injection mechanism after the 1 sprint ends.
-    companion object {
-        val movieProvider: MovieProvider by lazy {
-            MovieProvider()
-        }
-    }
-
```

Comment:
Movie provider singleton class - injected to the view model constructor as a singleton object gives possibility to handle all the methods connected with api data fetching